### PR TITLE
Check endpoint validity for pre-signed URLs

### LINKED ISSUE #1025: Can't get object with presigned url
1. Get presigned url:
```
COMMAND: ./neofs-s3-authmate  generate-presigned-url --endpoint 'localhost:41337' --method 'GET' --bucket '2fe0b189-2267-47af-86dc-2aaabdb510fe' --object 'temp_file_40227144-6d16-4247-a704-feeb3f89319d' --lifetime '30s' --aws-secret-access-key '38daa6c3d1a7767c9ee166b43600b4a14a259b04912a5456cc1439508d677d06' --aws-access-key-id 'DV3wySivBssfbrgg8KDdeGwyuesYuT2u25No5a5XLY3R0BB52CScxDCgVRMPg1rDDgw7Jg7CpErzKz7BUQbCRr9Bi'
RETCODE: 0

STDOUT:
{
  "URL": "localhost:41337/2fe0b189-2267-47af-86dc-2aaabdb510fe/temp_file_40227144-6d16-4247-a704-feeb3f89319d?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=DV3wySivBssfbrgg8KDdeGwyuesYuT2u25No5a5XLY3R0BB52CScxDCgVRMPg1rDDgw7Jg7CpErzKz7BUQbCRr9Bi%2F20241116%2F%2Fs3%2Faws4_request&X-Amz-Date=20241116T212214Z&X-Amz-Expires=30&X-Amz-SignedHeaders=host&X-Amz-Signature=f0b3d1d14e50ed021f6fb9aa3cdf17cfaf19a7b1cd8c7b0bb0eaea545a849949"
}
```
2. Try to download file under this url:

```
403 Forbidden
&lt;?xml version="1.0" encoding="UTF-8"?&gt;
&lt;Error&gt;&lt;Code&gt;SignatureDoesNotMatch&lt;/Code&gt;&lt;Message&gt;The request signature we calculated does not match the signature you provided. Check your key and signing method.&lt;/Message&gt;&lt;Key&gt;temp_file_40227144-6d16-4247-a704-feeb3f89319d&lt;/Key&gt;&lt;BucketName&gt;2fe0b189-2267-47af-86dc-2aaabdb510fe&lt;/BucketName&gt;&lt;Resource&gt;/2fe0b189-2267-47af-86dc-2aaabdb510fe/temp_file_40227144-6d16-4247-a704-feeb3f89319d&lt;/Resource&gt;&lt;RequestId&gt;bdc15d71-af6d-42c0-8668-4734f600dd2b&lt;/RequestId&gt;&lt;HostId&gt;ed5f5694-be01-4b97-82e3-149a0d5fcc8f&lt;/HostId&gt;&lt;/Error&gt;
```

Tried getting presigned url via authmate and s3 clients (aws s3 + boto3) - same error upon getting the object: `The request signature we calculated does not match the signature you provided`

Example - https://rest.fs.neo.org/HXSaMJXk2g8C14ht8HSi7BBaiYZ1HeWh2xnWPGQCg4H6/700-1731792182/index.html#suites/405fafaf31f575ac72014ad927d2fd6a/2f18eefdcf13a177/ (https://github.com/nspcc-dev/neofs-testcases/pull/896)

### PR DESCRIPTION
**Pull request by smallhive**
Closes #1025.

### CODE REVIEW

**Comment by smallhive** [comment]
I have a question. Do we need to have `generate-presigned-url` inside `neofs-s3-authmate`?

My concern is about the internal command:

```shell
./neofs-s3-authmate generate-presigned-url \
  --endpoint 'http://localhost:19080' \
  --method 'GET' --bucket 'heh1732164910' \
  --object 'file-1732164943.txt' \
  --lifetime '3600s' \
  --aws-secret-access-key '48966f87c37c12c15548a30ef3c1319438b9364a11ca033bea6fc4e2eb5565de' \
  --aws-access-key-id '78Tx3pRmSXNryyVUA7N1WCxg8JZWaGcR2G5M8AWbWuur04iSdjuV8LM51duiiSgY8KqyuMTx4XFLEMfWPtC9fkWLo'
```
It easily can be replaced with native AWS CLI:
```
aws s3 presign s3://heh1732164910/file-1732164943.txt --expires-in 3600 --endpoint-url localhost:19080
```

and leads to the same result. Of course, `aws configure` used the same `access` and `secret` keys.

Don't you mind if we remove this `generate-presigned-url` from `neofs-s3-authmate` in the separate issue?
@roman-khimov

File: api/auth/center.go [hunk 1]
```diff
@@ -314,7 +315,7 @@ func cloneRequest(r *http.Request, authHeader *authHeader) *http.Request {
 
 func (c *center) checkSign(authHeader *authHeader, box *accessbox.Box, request *http.Request, signatureDateTime time.Time) error {
 	awsCreds := credentials.NewStaticCredentials(authHeader.AccessKeyID, box.Gate.AccessKey, "")
-	signer := v4.NewSigner(awsCreds)
+	signer := v4amz.NewSigner(awsCreds)
```
Comment (roman-khimov) [inline api/auth/center.go:318]: Why it was added initially?

Comment (smallhive) [reply api/auth/center.go:318]: Our implementation?

Comment (roman-khimov) [reply api/auth/center.go:318]: Yes.

Comment (roman-khimov) [reply api/auth/center.go:318]: https://github.com/nspcc-dev/neofs-s3-gw/issues/528

Comment (roman-khimov) [reply api/auth/center.go:318]: Maybe https://github.com/nspcc-dev/neofs-s3-gw/issues/1028 can help here.

**Comment by evgeniiz321** [comment]
@roman-khimov nope, i don't need the authmate generate-presigned-url. Using aws cli/boto3 seems to be the right way to go.

**Comment by smallhive** [comment]
I left only a fix for presign URLs. Internal AWS implementation displaced to another PR

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**